lib/handler.js

```diff
@@ -22,13 +22,15 @@ const errorLog = e => process.env.NODE_ENV === 'development' && console.error(e)
 
 Handler.prototype.createAuth = function createAuth () {
   const {
-    oauthHost,
     authorizationPath = '/authorize',
     accessTokenPath = '/token',
-    oauthClientID: clientId,
-    oauthClientSecret: clientSecret,
     scopes = []
   } = this.opts
+
+  const oauthHost = typeof this.opts['oauthHost'] === 'function' ? this.opts['oauthHost'](this.req) : this.opts['oauthHost']
+  const clientSecret = typeof this.opts['oauthClientSecret'] === 'function' ? this.opts['oauthClientSecret'](this.req) : this.opts['oauthClientSecret']
```

Comment:
why are we moving this to per request, can't we just use this with the original config?

Reply:
Due to kube probe requests, these methods gets invoked all the time. As soon the drone build is completed nuxt server gets a request(probe) from kube pods and oauthHost property gets replaced with a string or whatever we return from oauthHost method which we assign in nuxt.config and when we click on sign-in button actual request goes to nuxt server from browser and this time oauthHost is not a function to invoke and correct URL can't be generated at runtime.